### ISSUE TITLE
hslua broken on i686

### ISSUE THREAD

**Issue by woffs** [NixOS/nixpkgs#39470]
## Issue description
compilation of haskellPackages.hslua fails with one failed test and breaks git-annex and pandoc which depend on that

### Steps to reproduce
```
[woffs@lapdoepp:~/git/nixpkgs] (master=) $ nix-build -A haskellPackages.hslua
[...]
    C functions
      Registering a C function and calling it from Lua:           FAIL
        test/Foreign/LuaTest.hs:162:
        greeting function failed
        expected: Right ["Caffeine","induced","nonsense"]
         but got: Right []
[...]
1 out of 113 tests failed (1.83s)
Test suite test-hslua: FAIL
```

## Technical details
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"i686-linux"`
 - host os: `Linux 4.14.35, NixOS, 18.03.132083.06c576b0525 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(woffs): `""`
 - channels(root): `"nixos-18.03.132083.06c576b0525, nixos1709-17.09.3238.e984f9e48e1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by Anton-Latukha** [comment]
@woffs 

Now it builds normally: https://hydra.nixos.org/job/nixpkgs/trunk/haskellPackages.hslua.i686-linux

**Comment by woffs** [comment]
Thank you @Anton-Latukha, it seems to work in NixOS now, and still fails in Nix on Debian.

**Comment by woffs** [comment]
Mysteriously, it is broken again with nixos-18.03 reaching 68e02f8ff21. It was ok at 14c248a4ab7. And it was broken before, as stated in the issue description, at 06c576b0525

**Comment by woffs** [comment]
Bisecting shows:
- It was fist broken at 9f30660e02102af6ca2976048d2ecea3a3e35639, which was a rather big hackage update @peti,
- it was fixed around 30ff9ca1958 and ab6d08132daea7cfc8437027f156d6bf9f64941a, which are python and python-cffi upgrades,
- the new breakage was introduced by 47c09dff6232fe7e4f418fe649ec4483915c1609, which is a sqlite update

**Comment by woffs** [comment]
It's a very strange and spurious error. I managed to build successfully and am closing this again for now.
